### ISSUE TITLE
Design and implement the architecture of our analyzer

### ISSUE THREAD

**Issue by SebastianZimmeck** [privacy-tech-lab/privacy-pioneer#12]
As we have now settled on Django/Python, Selenium + BrowserMob, and Heroku, the question becomes how to design and implement our analyzer. I have updated the [architectural overview](https://github.com/privacy-tech-lab/integrated-privacy-analysis/blob/master/README.md). 

Essentially, we could design our analyzer as a web app that the developer logs in to and runs on the browser of our web app their web app, which is then being analyzed by us. So, everything could work remotely. The developer does not need to install anything locally on their end. As @davebaraka mentioned, maybe it is more clear to actually run two servers: one where the developer is running their web app and one with our analysis logic.

An alternative setup could be that only our analysis module runs on the server and the developer is running their app locally. Their app would then communicate with our server on the current running analysis, results to display, etc. Maybe, in this architecture we ask the developer to integrate into their web app an SDK that we provide that communicates with our sever and drives the analysis.

The bottom line is to come up with an architecture that does not require the developer to do a whole lot of set up steps on their end and also make our analysis work on different platforms (e.g., independently of whether the developer is using Windows or macOS locally).
